src/group_theory/perm/cycles.lean

```diff
@@ -313,6 +361,31 @@ lemma is_cycle.same_cycle {f : perm β} (hf : is_cycle f) {x y : β}
   (hx : f x ≠ x) (hy : f y ≠ y) : same_cycle f x y :=
 hf.exists_gpow_eq hx hy
 
+lemma same_cycle.nat' [fintype β] {f : perm β} {x y : β} (h : same_cycle f x y) :
+  ∃ (i : ℕ) (hpos : 0 < i) (h : i ≤ order_of f), (f ^ i) x = y :=
```

Comment:
Any reason for this phrasing and not the simpler:
```suggestion
  ∃ (i : ℕ) (h : i < order_of f), (f ^ i) x = y :=
```
If you need your version, can you prove it in terms of this version?